runtime-common/error.ts

```diff
@@ -1,60 +1,168 @@
+import { getReasonPhrase } from "http-status-codes";
+export interface ErrorDetails {
+  status?: number;
+  title?: string;
+  source?: {
+    pointer?: string;
+    header?: string;
+    parameter?: string;
+  };
+}
+
 export class CardError extends Error {
-  constructor(readonly response: Response) {
-    super(`CardError ${response.status}`);
+  detail: string;
+  status: number;
+  title?: string;
+  source?: ErrorDetails["source"];
+  isCardError: true = true;
+  additionalErrors: (CardError | Error)[] | null = null;
```

Comment:
This is our serializable CardError from the @cardstack/core workspace. I had to get more serious about our errors, such that I could serialize/deserialize the CardErrors over HTTP boundaries as they need to flow thru the loader now. We still have another ticket to serialize these into the search doc and fine tune them so the stack traces make sense. but the hard part should hopefully be done now